src/main/java/com/salesforce/dataloader/security/EncryptionAesUtil.java

```diff
@@ -0,0 +1,198 @@
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * com.salesforce.dataloader.security
+ *
+ * @author xbian
+ */
+public class EncryptionAesUtil {
+
+    private static final Logger LOGGER = Logger.getLogger(EncryptionAesUtil.class);
+
+    // Support single text encryption and decryption
+
+    static public Cipher cipher;
+    static public byte[] cipherKey;
+
+
+    public static final int ENCRYPTION_KEY_LENGTH_IN_BYTES = 16;
+    public static final String DEFAULT_KEYFILE_NAME = "dataLoader.key";
+
+    static {
+        try {
+            cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+
+        } catch (Exception e) {
+            LOGGER.error("Error to initialize Encryption: " + e.getMessage());
+            throw new  RuntimeException("Error to initialize Encryption: ", e);
+        }
+    }
+
+
+    private byte[] extractIvBytes(byte[] cipheredText) {
+        byte[] iv = new byte[ENCRYPTION_KEY_LENGTH_IN_BYTES];
+        System.arraycopy(cipheredText, 0, iv, 0, ENCRYPTION_KEY_LENGTH_IN_BYTES);
+        return iv;
+    }
+
+    private byte[] extractCipherContent(byte[] cipheredText) {
+        byte[] cipheredPassword = new byte[cipheredText.length - ENCRYPTION_KEY_LENGTH_IN_BYTES];
+        System.arraycopy(cipheredText, ENCRYPTION_KEY_LENGTH_IN_BYTES, cipheredPassword, 0, cipheredText.length - ENCRYPTION_KEY_LENGTH_IN_BYTES);
+        return cipheredPassword;
+
+    }
+
+    private byte[] concatenateByteArray(byte[] a, byte[] b) {
+
+        byte[] c = new byte[a.length + b.length];
+        System.arraycopy(a, 0, c, 0, a.length);
+        System.arraycopy(b, 0, c, a.length, b.length);
+        return c;
+    }
+
+    public byte[] generateEncryptionKey() {
+        // The secure random uses threadId and tick to make sure it is not repeatable
+        SecureRandom random = new SecureRandom();
+        byte bytes[] = new byte[ENCRYPTION_KEY_LENGTH_IN_BYTES];
+        random.nextBytes(bytes);
+        return bytes;
+    }
+
+    public String createUserProfileKeyName() {
+        String path = System.getProperty("user.home") + File.separator + "dataloader";
+        File customDir = new File(path);
+
+        if (customDir.exists()) {
+            LOGGER.info(customDir + " exists");
+        } else if (customDir.mkdirs()) {
+            LOGGER.info(customDir + " was created");
+        } else {
+            LOGGER.info(customDir + " was not created");
+            throw new RuntimeException("Cannot create keyFile in directory:" + path);
+        }
+        return path + File.separator + DEFAULT_KEYFILE_NAME;
+    }
+
+    public String createKeyFile(String filePath) throws IOException {
+        if (filePath == null || filePath.isEmpty()) {
+            filePath = createKeyFile(createUserProfileKeyName());
+        }
+        byte[] key = generateEncryptionKey();
+        try (FileOutputStream fos = new FileOutputStream(filePath)) {
+            fos.write(key);
+        }
```

Comment:
Why not `catch` for IOException within this method instead of bubbling up?

Reply:
Catch means you need to handle this failure by using return false to bubble up the error, which is more handling in this case.